types/operations/comparisons/Comparison-inl.hpp

```diff
@@ -0,0 +1,664 @@
+
+template <bool left_nullable, bool right_nullable>
+TupleIdSequence* UncheckedComparator::compareColumnVectorsDefaultImpl(
+    const ColumnVector &left,
+    const ColumnVector &right,
+    const TupleIdSequence *filter,
+    const TupleIdSequence *existence_bitmap) const {
+  return InvokeOnColumnVector(
+      left,
+      [&](const auto &left_column_vector) -> TupleIdSequence* {  // NOLINT(build/c++11)
+    return InvokeOnColumnVector(
+        right,
+        [&](const auto &right_column_vector) -> TupleIdSequence* {  // NOLINT(build/c++11)
+#ifdef QUICKSTEP_ENABLE_VECTOR_PREDICATE_SHORT_CIRCUIT
+      static constexpr bool short_circuit = true;
+#else
+      static constexpr bool short_circuit = false;
+#endif
+      DCHECK_EQ(left_column_vector.size(), right_column_vector.size());
+      DCHECK((existence_bitmap == nullptr)
+             || (existence_bitmap->numTuples() == left_column_vector.size()));
+      DCHECK((filter == nullptr)
+             || ((existence_bitmap == nullptr) ? (filter->length() == left_column_vector.size())
+                                               : (filter->length() == existence_bitmap->length())));
+      TupleIdSequence *result = new TupleIdSequence(
+          (existence_bitmap == nullptr) ? left_column_vector.size()
+                                        : existence_bitmap->length());
+      if (short_circuit && (filter != nullptr)) {
+        if (existence_bitmap != nullptr) {
+          TupleIdSequence::const_iterator existence_it = existence_bitmap->begin();
+          for (std::size_t cv_pos = 0;
+               cv_pos < left_column_vector.size();
+               ++cv_pos) {
+            DCHECK(existence_it != existence_bitmap->end());
+            result->set(*existence_it,
+                        filter->get(*existence_it)
+                            && this->compareDataPtrs(
+                                left_column_vector.template getUntypedValue<left_nullable>(cv_pos),
+                                right_column_vector.template getUntypedValue<right_nullable>(cv_pos)));
+            ++existence_it;
+          }
+          DCHECK(existence_it == existence_bitmap->end());
+        } else {
+          for (TupleIdSequence::const_iterator filter_it = filter->begin();
+               filter_it != filter->end();
+               ++filter_it) {
+            result->set(*filter_it,
+                        this->compareDataPtrs(
+                            left_column_vector.template getUntypedValue<left_nullable>(*filter_it),
+                            right_column_vector.template getUntypedValue<right_nullable>(*filter_it)));
+          }
+        }
+      } else {
+        if (existence_bitmap != nullptr) {
+          TupleIdSequence::const_iterator existence_it = existence_bitmap->begin();
+          for (std::size_t cv_pos = 0;
+               cv_pos < left_column_vector.size();
+               ++cv_pos) {
+            DCHECK(existence_it != existence_bitmap->end());
```

Comment:
This file is a refactoring of the original code in `Comparison.cpp`. But I think it makes sense to remove the check because at the beginning we already have this:
```
      DCHECK((existence_bitmap == nullptr)
             || (existence_bitmap->numTuples() == left_column_vector.size()));
```
Will update it.